src/integrant/core.cljc

```diff
@@ -24,16 +24,28 @@
     (ref? v)  (list (:key v))
     (coll? v) (mapcat find-refs v)))
 
+(defn find-derived
+  "Return a seq of all entries in a map, m, where the key is derived from the
+  keyword, k. If there are no matching keys, nil is returned."
+  [m k]
+  (seq (filter #(isa? (key %) k) m)))
+
 (defn dependency-graph
-  "Return a dependency graph of all the refs in a config."
+  "Return a dependency graph of all the refs in a config. Resolve derived
+  dependencies."
   [config]
-  (reduce-kv (fn [g k v] (reduce #(dep/depend %1 k %2) g (find-refs v)))
+  (reduce-kv (fn [g k v]
+               (->>
+                (find-refs v)
+                (mapcat (partial find-derived config))
```

Comment:
Right, that's fine for now. I might decide to use a function like `find-derived-exactly-1`, but I was also thinking about adding in a `valid?` function that determines if a map is a valid configuration. Then I could add it as a precondition. Alternatively, once Clojure 1.9 is out, or at least in beta, perhaps a spec could be used instead. There are a few ways to go about it, which is why I want to delay that decision for now.